### ISSUE TITLE
SqlContext.createDataFrame should support JavaScript object in addition to StructType

### ISSUE THREAD

**Issue by billreed63** [EclairJS/eclairjs-nashorn#250]
Can we enable DataFrame creation usig JavaScript object?
In Java the user cane create a DataFrame using a JavaBean

```
var localTraining = [
      new LabeledDocument(0 , "a b c d e spark", 1.0),
      new LabeledDocument(1 , "b d", 0.0),
      new LabeledDocument(2 , "spark f g h", 1.0),
      new LabeledDocument(3 , "hadoop mapreduce", 0.0)];
    var training = sqlContext.createDataFrame(sc.parallelize(localTraining), LabeledDocument.class);
```

Can we provide similar function using JavaScript object instead of JavaBean?

**Comment by billreed63** [comment]
@pberkland

**Comment by pberkland** [comment]
attempting to implement something like

```
var training = sqlContext.createDataFrameFromJson(sc.parallelize(localTraining), {
                id : "Integer",
               text: "String",
               label: "String"
             })
```

The function will probably have to different name than 'createDataFrame' so the there is less ambiguity.

**Comment by pberkland** [comment]
fixed by commit 401c27e5b80e4a246f65f1065e4427f51c37321d
